pkg/apis/extensions/register.go

```diff
@@ -47,6 +47,11 @@ var (
 
 // Adds the list of known types to api.Scheme.
 func addKnownTypes(scheme *runtime.Scheme) error {
+	// Add known resource names introduced in this API
+	api.AddStandardResourceNames(ResourceDeployments, ResourceReplicaSets)
```

Comment:
agree, I will populate the string sets in register using this function for no-name group resources.